### ISSUE TITLE
Snapshots is not working when using native ESM module loader

### ISSUE THREAD

**Issue by trygve-lie** [tapjs/tapjs#687]
It seems like snapshots is not working when using the native ESM loader in node when the module being tested is an ESM.

My module is defined as a ESM by defining `type` as `module` in `package.json`:

```json
{
    "name": "foo",
    "version": "1.0.0",
    "type": "module"
}
```
My module and tests are then written as ESM. I run the tests with the `--no-esm` flag to use the native ESM module loader in later node.js (12 and 14).

Generating snapshots by the following command works well:

```sh
tap --no-esm tests/*.js --snapshot
```

Snapshots are written to the `tap-snapshots` folder but when I run the tests without generating the snapshots with the following command I get an error:

```sh
tap --no-esm tests/*.js
```
The error it outputs is (and a bit miss leading since the file it say it does not find does exist):

```sh
Snapshot file not found: /pah/to/module/tap-snapshots/my-file.js-TAP.test.js 
Run with TAP_SNAPSHOT=1 in the environment to create snapshot files
```

The real error message is actually [eaten by tap](https://github.com/tapjs/node-tap/blob/master/lib/snapshot.js#L28). The real problem here is as follow:

```sh
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /pah/to/module/tap-snapshots/my-file.js-TAP.test.js 
require() of ES modules is not supported.
require() of /pah/to/module/tap-snapshots/my-file.js-TAP.test.js  from /pah/to/module/node_modules/tap/lib/snapshot.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename my-file.js-TAP.test.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /pah/to/module/package.json.
```

In other words; when the module is defined as a ESM by setting `type` in `package.json`, the loader expects the snapshots to be ESM too due to its .js extension.

Renaming the snapshot file to `.cjs` as suggested by node does not solve the issue. Tap will then complain about the snapshot not existing.

Only workaround I have found to get this working is to put a `package.json` in the `tap-snapshots` folder without a `type` property. That makes the native ESM loader in node treat the snapshots as a CommonJS module.

**Comment by coreyfarrell** [comment]
This is a known issue, the workaround is to create `tap-snapshots/package.json` containing `{"type": "commonjs"}`.  The snapshot files will have a .cjs extension in tap v15 but that switch is a breaking change.

**Comment by trygve-lie** [comment]
Thanks for your quick reply.

If the solution for this is using a .cjs extension in tap v15 and no work is done on this yet I'm happy to take a stab at a PR for this.

**Comment by coreyfarrell** [comment]
This has already been changed in the v15 branch.  No ETA yet on v15 release.

**Comment by schickling** [comment]
Given v15 is out by now, can this issue be closed?

**Comment by rkesters** [comment]
I'm still seeing this on V16.0.0 with typescript. 

`npx tap  --snapshot --ts`
`
test/index.spec.ts 2> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file
extension ".ts" for /Users/rkesters/code/github/remark-pdf/test/index.spec.ts
`

**Comment by isaacs** [comment]
@rkesters Different issue. You're describing #807
